src/components/provider/PyatchProvider.jsx

```diff
@@ -34,6 +36,52 @@ const PyatchProvider = props => {
   let [spriteSize, setSpriteSize] = useState(100);
   let [spriteDirection, setSpriteDirection] = useState(90);
 
+  let [errorList, setErrorList] = useState([]);
+
+  //returns array with each line of code for given sprite id
+  pyatchEditor.getCodeLines = (sprite) => {
+    let linesOfCode = [];
+    let prev = 0;
+    if(!pyatchEditor.editorText[sprite]){
+      return [];
+    }
+    for(let i = 0; i < pyatchEditor.editorText[sprite].length; i++){
+      if(pyatchEditor.editorText[sprite][i]=='\n'){
+        linesOfCode.push(pyatchEditor.editorText[sprite].substring(prev, i));
+        prev = i+1;
+      }
+    }
+    linesOfCode.push(pyatchEditor.editorText[sprite].substring(prev, pyatchEditor.editorText[sprite].length));
+    return linesOfCode;
+  }
+
+
+  function generateError(error){
+    let line = error.line;
+    let sprite = error.sprite;
+    let linesOfCode = pyatchEditor.getCodeLines(sprite);
+    let priorText = [];
+    let afterText= [];
+    for (let i=3; i>0; i--){
+      if(line-i-1>=0 && line-i-1<linesOfCode.length)priorText.push(line-i+" "+linesOfCode[line-i-1]);
```

Comment:
Can you please refactor the values `line-i-1`, `line-i+3`, and `line-i+4` into their own variables with relevant names? It would improve readability.